### PR TITLE
Add feature to resolve the spl-token v2 multisig bug

### PR DESCRIPTION
**Pull request by mvines**
Add mechanism that'll allow us to fix the spl-token v2 multisig issue across the existing clusters

### CODE REVIEW

**Comment by mvines** [comment]
@CriesofCarrots / @jackcmay - I just need review on the top commit please, the rest are covered elsewhere

**Comment by jackcmay** [comment]
Instead of reviewing the top commit, how about directly review: https://github.com/solana-labs/solana/pull/12376?

**Comment by mvines** [comment]
> Instead of reviewing the top commit, how about directly review: #12376?

Sure!

**Review by CriesofCarrots — APPROVED** [review]
Top commit lgtm!

**Comment by mvines** [comment]
Rebased this PR on #12454, I'll wait until that PR lands though before bringing this PR out of draft

**Comment by mvines** [comment]
Depends-on: #12502

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -3576,6 +3580,15 @@ impl Bank {
         self.set_compute_budget(compute_budget);
     }
 
+    fn apply_spl_token_v2_multisig_fix(&mut self) {
+        if let Some(mut account) = self.get_account(&inline_spl_token_v2_0::id()) {
+            self.capitalization.fetch_sub(account.lamports, Relaxed);
+            account.lamports = 0;
+            self.store_account(&inline_spl_token_v2_0::id(), &account);
+            self.remove_executor(&inline_spl_token_v2_0::id());
```
Comment (mvines) [inline runtime/src/bank.rs:3588]: @jackcmay - thanks, `remove_executor()` slots in nicely here

File: runtime/src/bank.rs [hunk 2]
```diff
@@ -9046,4 +9059,27 @@ mod tests {
...
+            &Account {
+                lamports: 100,
+                ..Account::default()
+            },
+        );
+        assert_eq!(bank.get_balance(&inline_spl_token_v2_0::id()), 100);
+        let original_capitalization = bank.capitalization();
+
+        bank.apply_spl_token_v2_multisig_fix();
+
```
Comment (mvines) [inline runtime/src/bank.rs:9080]: TODO: Also confirm here that token is no longer in the executor cache...

File: runtime/src/bank.rs [hunk 3]
```diff
@@ -3662,6 +3666,15 @@ impl Bank {
         self.set_compute_budget(compute_budget);
     }
 
+    fn apply_spl_token_v2_multisig_fix(&mut self) {
+        if let Some(mut account) = self.get_account(&inline_spl_token_v2_0::id()) {
+            self.capitalization.fetch_sub(account.lamports, Relaxed);
```
Comment (ryoqun) [inline runtime/src/bank.rs:3671]: :eyes: (Thank for keeping cap. in sync)

**Pull request merged**